File: python/lv1/add_plus_minus.py
# 음양 더하기

'''
Answer 1.

def solution(absolutes, signs):
    answer = 0
    
    for i in range(len(absolutes)):
        if signs[i] == False:
            answer -= absolutes[i]
        else:
            answer += absolutes[i]
    
    return answer
'''

'''
Answer 2.

def solution(absolutes, signs):
    answer=0
    
    for absolute,sign in zip(absolutes,signs):
        if sign:
            answer += absolute
        else:
            answer -= absolute
    return answer
'''

'''
Answer 3.

def solution(absolutes, signs):
    return sum(absolutes if sign else -absolutes for absolutes, sign in zip(absolutes, signs))
'''

File: cpp/lv1/add_plus_minus.cpp
// 음양 더하기

#include <string>
#include <vector>

using namespace std;

/* Answer 1.
  
int solution(vector<int> absolutes, vector<bool> signs) {
    int answer = 0;
    
    for(int i = 0; i < absolutes.size(); i++)
    {
        if(signs[i] == true) //signs[i]로만 작성 가능
        {
            answer += absolutes[i];
        }
        else
        {
            answer -= absolutes[i];
        }
    }
    return answer;
} */

/* Answer 2.

int solution(vector<int> absolutes, vector<bool> signs) {
    int answer = 0;
  
    for(int i = 0; i < signs.size(); i++)
    {
        if(!signs[i]) absolutes[i] *= -1;
        answer += absolutes[i];
    }
    return answer;
} */
